src/player.cpp

```diff
@@ -12241,14 +12241,14 @@ int player::encumb( body_part bp ) const
 int player::encumb(body_part bp, double &layers, int &armorenc) const
 {
     int ret = 0;
-    int layer[MAX_CLOTHING_LAYER] = { };
+    double layer[MAX_CLOTHING_LAYER] = { };
 
     for( auto& w : worn ) {
         if( !w.covers(bp) ) {
             continue;
         }
 
-        layer[w.get_layer()] += 10;
+        layer[w.get_layer()] += 10.f * (w.get_coverage() / 100);
```

Comment:
This does not work as you expect it to work. `item::get_coverage` returns an `int`, so the division is a simple integer division, which discards the remainder. It's only converted to `float` afterwards for the multiplication.

In other words: whether the multiplication is with `10` (`int`) or with `10.f` (float) does not matter, the result has the same numeric value (albeit a different type).

Also note that `item::get_coverage` is in percent, so it's usually below 100. The integer division will yield 0 for the coverage values from 0 to 99. Only if the coverage is 100, the division yields 1. To make this work with floating point, you need to convert at least on operand of the division to float, e.g. `(item::get_coverage() / 100.0f)`